.github/workflows/build.yml

```
@@ -0,0 +1,93 @@
+name: Build
+
+# Run on push to main and any pull request.
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

```suggestion
  push:
    tags:
      - v*
```

this workflow uses a secret and publish some elements. 

and I recommend to restore my previous file `go-cross.yml`

---

Thanks for the quick feedback! What should be different in the PR build process versus release? Is the OS matrix in go-cross intended to verify build success for common platforms, whereas the release output can come from a single build platform?

---

The release will publish artifacts (archive, GitHub release, Docker images, etc.), a PR will not do that.
A PR doesn't need to build Docker images or create archives.
It's better to not create one workflow with the conditions but to have a dedicated workflow by topic.

In general, nobody should push on a repo without creating a PR before, so the workflow of the PR will check tests, linting, and binary builds.
So when you merge a PR, tests, and linting are useless because they have been already checked.

This is the same thing for builds: the build for a release is not just binary but a set of builds specific to a release (archive, GitHub release, Docker images, etc.).